### PR TITLE
feat: add NUMA performance collector

### PR DESCRIPTION
**Pull request by jra3**
## Summary
- Add new performance collector for monitoring NUMA (Non-Uniform Memory Access) statistics
- Provides per-node memory metrics and topology information for multi-socket systems
- Follows established collector patterns with comprehensive tests

## Changes
- **New NUMA collector** (`pkg/performance/collectors/numa.go`)
  - Monitors per-node memory allocation counters (hits, misses, foreign allocations)
  - Tracks per-node memory usage (total, free, used, file/anon pages)
  - Collects NUMA topology (CPU assignments, inter-node distances)
  - Reports system NUMA configuration (auto-balancing status)

- **Comprehensive test suite** (`pkg/performance/collectors/numa_test.go`)
  - Constructor validation tests
  - Tests for various NUMA configurations (non-NUMA, single node, multi-node)
  - Error handling and malformed data tests
  - Edge case coverage (whitespace tolerance, CPU list parsing)

- **Type definitions** in `pkg/performance/types.go`
  - Added `NUMAStats` and `NUMANodeStats` structures
  - Added `MetricTypeNUMA` to metric types

## Implementation Details
- Critical files (numastat) cause node to be skipped if missing/invalid
- Optional files (meminfo, cpulist, distance) log warnings but continue gracefully
- Monotonically increasing counters for allocation statistics
- Gauge metrics for memory usage that fluctuate

## Test Plan
- [x] Unit tests pass (`go test ./pkg/performance/collectors/ -run TestNUMACollector`)
- [x] Linting passes (`make lint`)
- [x] Code formatted (`make fmt`)
- [ ] Test on actual NUMA system
- [ ] Verify metrics are collected correctly in agent

### CODE REVIEW

File: pkg/performance/collectors/numa.go [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+
+func NewNUMACollector(logger logr.Logger, config performance.CollectionConfig) (*NUMACollector, error) {
+	// Validate that HostSysPath is absolute
+	if !filepath.IsAbs(config.HostSysPath) {
+		return nil, fmt.Errorf("HostSysPath must be an absolute path, got: %q", config.HostSysPath)
+	}
+	// Validate that HostProcPath is absolute
+	if !filepath.IsAbs(config.HostProcPath) {
+		return nil, fmt.Errorf("HostProcPath must be an absolute path, got: %q", config.HostProcPath)
+	}
```
Comment (haq204) [inline pkg/performance/collectors/numa.go:71]: This probably should use the new central validation?

**Review by haq204 — APPROVED** [review]
